py3status/modules/whatismyip.py

```diff
@@ -61,43 +74,38 @@ def on_click(self, event):
         else:
             self.mode = 'ip'
 
-    def _get_my_ip_and_location(self):
+    def _get_my_ip_info(self):
         """
         """
         try:
-            if self.py3.format_contains(self.format, 'country'):
-                resp = urlopen(self.url_geo, timeout=self.timeout).read()
-                resp = json.loads(resp)
-                country = resp['country']
-                ip = resp['query']
-            else:
-                country = None
-                ip = urlopen(self.url, timeout=self.timeout).read()
-                ip = ip.decode('utf-8')
+            resp = self.py3.request(self.url_geo, timeout=self.timeout).json()
+            info = {}
+            for placeholder in self.py3.get_placeholders_list(self.format):
+                if placeholder in resp:
+                    info[placeholder] = resp[placeholder]
+            for old, new in self.substitutions.items():
+                info[old] = resp.get(new)
+            return info
         except Exception:
-            country = None
-            ip = None
-        return country, ip
+            return None
```

Comment:
My guess. `try/except self.Py3.RequestException:` the resp (Line 81). Leave everything else exposed. If something potentially happened, py3status should stop the module, turn the module red, pass the exception to the red `whatismyip` module where you can click on it and it'll expand the exception/error message... That and one can find the issue in the log. From there, one can figure out what the issue was... 

It's possibly better than keep requesting API only to get something always failed in the json/placeholder parser... and still continue returning None... which would keep this loop going.